streaming/src/main/scala/org/apache/gearpump/streaming/dsl/Stream.scala

```diff
@@ -115,20 +121,17 @@ class Stream[T](
    *
    * For example,
    * {{{
-   * Stream[People].groupBy(_.gender).flatmap(..).filter.(..).reduce(..)
+   * Stream[People].groupBy(_.gender).flatMap(..).filter(..).reduce(..)
    * }}}
    *
-   * @param fun  Group by function
+   * @param fn  Group by function
    * @param parallelism  Parallelism level
    * @param description  The description
    * @return  the grouped stream
    */
-  def groupBy[Group](fun: T => Group, parallelism: Int = 1, description: String = null)
-    : Stream[T] = {
-    val groupOp = GroupByOp(fun, parallelism, Option(description).getOrElse("groupBy"))
-    graph.addVertex(groupOp)
-    graph.addEdge(thisNode, edge.getOrElse(Shuffle), groupOp)
-    new Stream[T](graph, groupOp)
+  def groupBy[GROUP](fn: T => GROUP, parallelism: Int = 1, description: String = "groupBy")
+    : GroupByStream[T, GROUP] = {
+    groupByWindow(DefaultGroupBy(fn), parallelism, description)
```

Comment:
Seems odd: groupBy -> groupByWindow -> GroupByStream. Suggests that all groupBy's have windowing semantics.